### PR TITLE
Replace registry repository column

### PR DESCRIPTION
**Pull request by angelina-momin**
Resolves issue: [#88](https://github.com/codecheckers/register/issues/88)
Related PR for register repo: Havent opened yet because I need to fix the codecheck.yml files.

Changes made:

- Bumped version to `0.9.0`
-  In `utils_preprocess_register.R` added function `add_paper_links`. For `codecheck.yml` has extra text in the paper reference field, the plain title is added with no hyperlink.
- Removed function `add_repository_links_html` from `utils_render_register_htmls.R`
- Removed function `add_repository_links_md` from `utils_render_register_mds.R`
- In `create_md_table` added code to only keep the columns specified in CONFIG
- Redid `filter_drop_register_columns` functions to now take in `file_type` as arg. This is used to determine the columns to keep
- Update `validate_codecheck_yml`- it now validates codecheck_yml$paper$reference. If the paper reference is not a link it throws an error.
- Update function documentation with ROxygen
- Adjusted files in `inst/yaml` according to the changes in `validate_codecheck_yml`.

As a result of the code changes, the following register files have changed

- `index.html`- Replace "register" column with "Paper Title"
- `register.md`- Identical to index.html. Replace "Register" column with "Paper Title"

New look of the register:
Note that certain paper tiles do not have hyper links because their `codecheck.yml` has extra text in the paper reference field. I will need to fix that manually in each of the `codecheck.yml` file

image

### CODE REVIEW

**Comment by angelina-momin** [comment]
@nuest  Could you review the code for now?  Please dont merge after you have approved. In the meanwhile i will fix the `codecheck.yml` files that have issues with the paper reference

The code already works and generates output as expected however I want to test again after fixing the `codecheck.yml` files

File: R/utils_preprocess_register.R
Comment (nuest) [inline R/utils_preprocess_register.R:29]: We can leave this as is now, but since I'm not afraid of dependencies for the current setup, we could also use the `stringr` package for this kind of task: https://stringr.tidyverse.org/reference/str_trim.html IMO that would be less error-prone and also more readable (for the regex-challenged like me).

Nothing to change now unless you're intrigued, but keep in mind for the future.

File: R/utils_preprocess_register.R
Comment (nuest) [inline R/utils_preprocess_register.R:38]: I believe this works, but I would prefer a plain if-else statement for readability.

File: R/utils_preprocess_register.R
Comment (nuest) [inline R/utils_preprocess_register.R:52]: Cool. Getting a hang of the tidy R world and piping!

Did you know that R also has a native pipe operator by now? `|>`

https://www.tidyverse.org/blog/2023/04/base-vs-magrittr-pipe/

File: R/utils_render_register_general.r
Comment (nuest) [inline R/utils_render_register_general.r:154]: You can also provide a default argument in the function definition:

```R
render_register <- function(register_table, table_details, filter, output_type = "other"){
```

This way you can simply pass on the `output_type`, and every time you need "other", you do not provide the final argument to the function.

File: inst/extdata/config.R
Comment (nuest) [inline inst/extdata/config.R:23]: "other" is basically HTML and MD then, right?

If other is capturing only these two cases, I would prefer a more explicit approach and list both here. OK for you? Am I missing something?

**Review by nuest — CHANGES_REQUESTED** [review]

File: R/utils_preprocess_register.R
Comment (angelina-momin) [reply R/utils_preprocess_register.R:53]: Ooh that's cool, didnt know that. Will keep note for new PRs

File: R/utils_render_register_general.r
Comment (angelina-momin) [reply R/utils_render_register_general.r:154]: Good point, completely missed that. Thanks!

File: inst/extdata/config.R
Comment (angelina-momin) [reply inst/extdata/config.R:23]: Yes "other" was just for md and html. I implemented the change

**Comment by angelina-momin** [comment]
Resolved all the comments! Thanks for your input. Found 5 other `codecheck.yml` files that needed their paper references fixed. Created PRs for those and I am waiting for them to be merged (I dont have write access) before rerunning rendering to do a final check. I'll merge this PR after that

**Comment by nuest** [comment]
@angelina-momin I think I merged all the updates to the codecheck-YMLs. Can you continue now?

**Comment by angelina-momin** [comment]
@nuest  These codecheck YMLs still need to be merged:
https://github.com/codecheckers/domestic.wastewater.variability/pull/1
https://github.com/langtonhugh/scope/pull/1

Btw the checks are failing because of the issues with the codecheck.ymls

**Comment by nuest** [comment]
Hm... Okay, will take a look at the config files.

Let's also create an issue about what should fail the tests, what should fail the rendering, and what not. Can you please do that as I am off today and don't want us to forget?

**Comment by nuest** [comment]
@angelina-momin The final to YMLs are fixed, afaict, but the tests still fail. Can you continue now with this locally?

**Pull request merged**
